documentation/how-to-use-malware-detection-backend.md

```diff
@@ -0,0 +1,45 @@
+# Using malware detection backend
+
+If you are developing a service where users can upload files, you will probably want to check if these files are safe or not.
+
+The malware detection app is here to give you several backend implementations and others can be added to match your requirements.
+
+## Global idea
+
+A backend should inherit from `lasuite.malware_detection.backends.base.BaseBackend` class and implements the abstract methods declare in it.
+The backend service then is responsible to analyse the file and give an answer by calling a defined callback.
+
+### Abstract methods
+
+- `def analyse_file(self, file_path: str, **kwargs) -> None:`: it is the entrypoint to analyse a file. This method return nothing and can analyse the file by using a celery task. The parameters to use:
+    - `file_path`: the path to access the file. This path must be accessible with the django default storage.
+    - `kwargs`: arbitrary kwargs usuqble by the user using the library. These kwargs will be used by the callback. You can for example pass a resource id or whatever usefull for you.
+
+### Callback
+
+The settings `LASUITE_MALWARE_DETECTION_CALLBACK` defined a function (full module with dot to the function). This function receives these parameters:
+    - `file_path` the `file_path` used in the `analyse_file` method
+    - `status` the analysis status (see `lasuite.malware_detection.enums.ReportStatus`). It can be `safe`, `unsafe` and `unknown`. In case of `unknwon` you are responsible to determine what you should do
+    - `error_info` a dict containing a `error` and `error_code` properties.
+    - `kwargs` the kwargs you pass to the `analyse_file` method
```

Comment:
I might not understand the full purpose of the callback:
- Should we use a setting or should the backend also provide the callback?
- I'm not sure we will need the same callback for all kind of analyzed files, maybe the callback function has to be provided as a keyword argument of `analyse_file` ? or the callback function should manage all the cases?

Reply:
You all probably better understand with the tests.